src/site/content/en/mini-apps/mini-app-programming-way/index.md

```diff
@@ -0,0 +1,74 @@
+In practice, too many times, there is a certain lock-in regarding a design system you need to buy in to when you use a component,
+or the component library is distributed in a way that it is all or nothing, but no individual components can be easily added to a project.
+There are, however, atomic components that you can use in isolation, or libraries like [generic-components](https://github.com/thepassle/generic-components)
+that are unstyled on purpose.
+Finding an using those seems like a good idea.
+
+## Model-view-viewmodel
+
+The [model–view–viewmodel](/mini-app-markup-styling-and-scripting/#markup-languages) (MVVM) architectural pattern—that facilitates the
+separation of the development of the graphical user interface (the view) via a markup language from
+the development of the back-end logic (the model)—means the view is not dependent on any specific model platform.
+While there are some documented [disadvantages](https://docs.microsoft.com/en-us/archive/blogs/johngossman/advantages-and-disadvantages-of-m-v-vm) of the pattern, in general it works really well for applications of the complexity of mini apps.
+It can shine especially with rich templating libraries (see [next chapter](mini-app-example-project/)).
+
+## Page-wise thinking
+
+Debugging mini apps shows that they are essentially multi-page applications (MPA).
+This has many advantages, like, for example, it allows for trivial routing and conflict-free per-page styling.
+People have [successfully applied MPA architectures](https://medium.com/elemefe/upgrading-ele-me-to-progressive-web-app-2a446832e509) to Progressive Web Apps.
+Thinking in pages also helps manage resources like each page's CSS and JavaScript files, and other assets like images and videos.
+Most importantly, building this way means you get code-splitting for free. Each page by definition strictly only loads what it needs to function.
+
+## Build process
+
+Mini apps have [no visible build process](/mini-app-project-structure-lifecycle-and-bundling/#the-build-process).
+On the web, modern build tools like [Snowpack](https://www.snowpack.dev/) leverage JavaScript's native
+[module system](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)
+(known as ESM) to avoid unnecessary work and stay fast no matter how big a project grows.
+While it is early days for technologies like [Web Bundles](/web-bundles/), it is something that can be easily added
+to the build process.
+
+## Powerful capabilities
+
+The web platform has gained many [new capabilities](/tags/capabilities/) recently.
+Access to [devices](/tags/devices/) via [Bluetooth](/bluetooth/), [USB](/usb/), [HID](/hid/), [serial](/serial/),
+and [NFC](/nfc/) is all possible now.
+Where mini apps run in WebViews and depend on a [JavaScript bridge](/mini-app-markup-styling-and-scripting/#javascript-bridge-api),
+on the web these powerful capabilities are available directly.
```

Comment:
This section seems a bit weird. The title of the page is "programming the mini app way". It's not "comparison of web and mini app programming". So it's strange to suddenly talk about the web versus mini apps. This comment also applies to the build process section that talks a lot about web bundling.

Reply:
The mini apps way of programming here means you don't program against an API provided by the JavaScript bridge, but against the browser API directly. That's why it's here. I'll add a sentence.